src/wp-includes/webfonts-api/class-wp-webfonts-registry.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Webfonts API: Webfonts Registry
+ *
+ * @package WordPress
+ * @subpackage Webfonts
+ * @since 5.9.0
+ */
+
+/**
+ * Webfonts Registry.
+ *
+ * This registry exists to handle all webfonts.
+ *
+ * It handles the following within the API:
+ *  - loads the bundled provider files into memory;
+ *  - registers each provider with the API by:
+ *       1. creating an instance (object);
+ *       2. storing it in-memory (by its unique provider ID) for use with the API;
+ *
+ * @since 5.9.0
+ */
+class WP_Webfonts_Registry {
+
+	/**
+	 * An in-memory storage container that holds all registered webfonts
+	 * for use within the API.
+	 *
+	 * Keyed by font-family.font-style.font-weight:
+	 *
+	 *      @type string $key => @type array Webfont.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @var array[]
+	 */
+	private $registered = array();
+
+	/**
+	 * Registration keys per provider.
+	 *
+	 * Provides a O(1) lookup when querying by provider.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @var string[]
```

Comment:
`$registry_by_provider` is an array of the provider with arrays of the registration_keys
```suggestion
	 * @var array[]
```